test/helpers.js

```diff
@@ -175,6 +175,39 @@
     })
   });
 
+  MockServiceWithErrorCodeMapping = function(mapping) {
```

Comment:
I don't think this new mocking method differs a lot from `MockService()` other than the model. I don't think this constructor is necessary. Why don't use the `MockServiceFromApi()` instead and supply a new API with with `awsQueryCompatible` trait?

Reply:
I have refactored to remove some code duplication. I am not able to use `MockServiceFromApi` directly.